airflow/jobs/local_task_job.py

```diff
@@ -127,7 +127,6 @@ def heartbeat_callback(self, session=None):
             self.task_runner.terminate()
             return
 
-        self.task_instance.refresh_from_db()
```

Comment:
`heartbeat_callback` is part of this sequence:
run > _execute > _check_and_change_state_before_execution > heartbeat > heartbeat_callback

TI was refreshed by in `_check_and_change_state_before_execution` so it seems that there's no reason to do this one more time in `heartbeat_callback`.

Reply:
But it will not refresh TI during health checks